esphome/components/esp32/gpio_esp32_h2.py

```diff
@@ -0,0 +1,25 @@
+from esphome.const import (
+    CONF_NUMBER,
+)
+
+import esphome.config_validation as cv
+
+
+def esp32_h2_validate_gpio_pin(value):
+    # ESP32-H2 not yet supported
+    if value > 0:
+        raise cv.Invalid("ESP32-H2 isn't supported yet")
+    if value < 0 or value > 21:
+        raise cv.Invalid(f"Invalid pin number: {value} (must be 0-21)")
+    return value
```

Comment:
```suggestion
    # ESP32-H2 not yet supported
    raise cv.Invalid("ESP32-H2 isn't supported yet")
```

0 would pass this validation, plus if it's not supported there's no point in the additional validation here.

Reply:
Still active

Reply:
Oops - missed that one thanks